packages/inter-protocol/src/vaultFactory/vault.js

```diff
@@ -418,10 +452,6 @@ export const prepareVault = (baggage, marshaller, zcf) => {
           const oldCollateral = self.getCollateralAmount();
 
           if (phase === Phase.ACTIVE) {
-            assertProposalShape(seat, {
-              give: { Minted: null },
-            });
```

Comment:
This asserts something that can't be known at interface guard time: proposal requirements given the state of the vault.

Without it, `given` on line 458 will be undefined which will cause this error,
```
"leftAmount" "[undefined]" must be a pass-by-copy record, not "undefined
```
That's a UX regression from (approx):
```
actual undefined did not match expected AmountShape
```
But that's not perfect either and if we really want to eliminate assertProposalShape then this shouldn't use it anyway.

Please add after `getProposal()`,
```js
given || Fail`closing an active vault requires a give`
```

Reply:
@erights I see new commits since this request. Not sure if you disagree or just haven't noticed it.

Reply:
Just haven't gotten to it yet.

Reply:
Ok, I did it here: https://github.com/Agoric/agoric-sdk/pull/7094

That's my take on the vaultFactory changes in this PR and the other changes have other reviewers so I'll remove myself here.

Reply:
Thx. Reverted change here in favor of #7094. Resolving.